### PR TITLE
Implementing bitcode.

### LINKED ISSUE #3449: Pure python fallback dependancy for intbitset.
Project Proposal: [Proposal](https://docs.google.com/document/d/14ThtbrNYY6-gWWghhb_r2KvxlNqouf64S0INswXMc4s/edit?usp=sharing)
Parent issue: https://github.com/nexB/scancode-toolkit/issues/3210
Linked PR: https://github.com/nexB/bitcode/pull/1
We plan to create a pure python fallback dependancy for intbitset.
To achieve this, we use binary operation to simulate set operations in an efficient manner.

### PR DESCRIPTION
**Pull request by 35C4n0r**
This PR Implements the pure python variant of intbitset.

Class:
- [x] intbitset

Methods:
- [x] add
- [x] clear
- [x] copy
- [x] difference
- [x] difference_update
- [x] discard
- [ ] extract_finite_list
- [ ] fastdump
- [ ] fastload
- [ ] get_allocated
- [ ] get_size
- [ ] get_wordbitsize
- [ ] get_wordbytesize
- [x] intersection
- [x] intersection_update
- [x] isdisjoint
- [x] issubset
- [x] issuprset
- [ ] is_infinite
- [ ] pop
- [x] remove
- [x] strbits
- [x] symetric_difference
- [x] symetric_difference_update
- [x] tolist
- [x] union
- [x] union_update
- [ ] update
- [ ] update_with_signs

### CODE REVIEW

File: src/bitcode/intbitset.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ahocode is a trademark of nexB Inc.
```
Comment (JonoYang) [inline src/bitcode/intbitset.py]:
```suggestion
# bitcode is a trademark of nexB Inc.

```

File: src/bitcode/intbitset.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ahocode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/ahocode for support or download.
```
Comment (JonoYang) [inline src/bitcode/intbitset.py]:
```suggestion
# See https://github.com/nexB/bitcode for support or download.

```

File: src/bitcode/intbitset.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            raise ValueError("Value can't be negative")
+        self.bitset |= 1 << value
+
+    def clear(self):
+        self.bitset = 0
+
+    def copy(self):
+        """ Return a shallow copy of a set. """
+        new = intbitset()
+        intbitset.bitset = self.bitset
```
Comment (JonoYang) [inline src/bitcode/intbitset.py]: Should this be `new.bitset = self.bitset`?

**Review by JonoYang — CHANGES_REQUESTED** [review]
@35C4n0r Just pointing out a few small things

**Comment by 35C4n0r** [comment]
@JonoYang, thanks for the review. I have added the changes, a few tests, and I also tried adding the github workflow to run the test.

**Comment by JonoYang** [comment]
@35C4n0r I've enabled Azure Pipelines for CI testing on the ahocode and bitcode repos

**Comment by 35C4n0r** [comment]
@JonoYang I have removed the github workflows, and i don't thin i need to make any changes in the azure-pipelines.yml
The PR is ready for review.

**Comment by 35C4n0r** [comment]
@JonoYang also please make a repo for lxml so that I can start working on it.

**Comment by JonoYang** [comment]
@35C4n0r I've created the xml fallback library repo at https://github.com/nexb/sanexml

**Comment by JonoYang** [comment]
@35C4n0r Is there anything else you need to add to this PR, otherwise I will merge it.

**Comment by 35C4n0r** [comment]
@JonoYang, I don't have anything more to add to PR.

**Comment by JonoYang** [comment]
@35C4n0r I will merge the PR. Thanks for the contributions!

**Pull request merged**